### PR TITLE
Add sign and verify

### PR DESCRIPTION
**Pull request by aljones15**
@msporny 

This puts both binary verify and sign functions into one binary.

For those just joining us this is the http signatures binary for the http signature spec tests.

### CODE REVIEW

**Comment by aljones15** [comment]
> 👍 looks good.
> I notice it uses Node native crypto - are there ambitions to use this lib in the browser?

there actually are so replacing this stuff with node forge is on the table. at the moment this at least gives us something to work with to test the http signatures spec. thanks for the review.

File: lib/index.js [hunk 1]
```diff
@@ -192,6 +130,113 @@ api.parseRequest = (request, options) => {
...
+/**
+ * Extracts the psedu headers(created) and (expires)
+ * from the parameters of a signature string.
+ *
+ * @param {Object} params - Signature parameters.
+ * @param {Object} headers - Request headers.
+ *
+ * @returns {Object} Headers.
+ */
+function extractPseudoHeaders(params, headers) {
```
Comment (aljones15) [inline lib/index.js]: @dlongley psuedo headers are extracted here too.

File: bin/http-signature-header-js [hunk 2]
```diff
@@ -4,89 +4,21 @@
  * Copyright (c) 2019 Digital Bazaar, Inc. All rights reserved.
 */
 
+// this module is the binary used by the http signature spec test stuies
```
Comment (dlongley) [inline bin/http-signature-header-js]:
```suggestion
// this module is the binary used by the http signature spec test suites
```

File: bin/util.js [hunk 3]
```diff
@@ -0,0 +1,261 @@
+const crypto = require('crypto');
+const httpSigs = require('../lib/');
+const fs = require('fs');
+const util = require('util');
+const httpMessageParser = require('http-message-parser');
+const jsprim = require('jsprim');
+
+// this can be use to await paths to private keys and other files.
+// also can get be used to dereference keyId in middleware
```
Comment (dlongley) [inline bin/util.js]: grammar check these comments

File: lib/index.js [hunk 4]
```diff
@@ -192,6 +130,113 @@ api.parseRequest = (request, options) => {
     }
 
   }
+  return parsed;
+}
+
+api.parseSignatureHeader = parseSignatureHeader;
+
+/**
+ * Extracts the psedu headers(created) and (expires)
```
Comment (dlongley) [inline lib/index.js]:
```suggestion
 * Extracts the pseudo headers (created) and (expires)
```

**Pull request merged**